src/tools.js

```diff
@@ -79,6 +79,16 @@ const bulletproofDecodeURIComponent = (encodedURIComponent) => {
     }
 };
 
+// Ports returned by `parseUrl` when port is not explicitly specified.
+// Values are based on node docs: https://nodejs.org/api/url.html#url_url_port
+const STANDARD_PORTS_BY_PROTOCOL = {
+    'ftp:': 21,
+    'http:': 80,
+    'https:': 443,
+    'ws:': 80,
+    'wss:': 443,
+};
```

Comment:
@valekjo Do you think worth adding socks default port?

https://tools.ietf.org/html/rfc1928

Reply:
I don't think it's necessary, unless we decide to add all standard ports. This covers ports that are "stripped out" by `new URL(url)`. Is that ok?

Reply:
Sounds good